bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/framework/EquinoxContainer.java

```diff
@@ -240,6 +256,12 @@ void close() {
 		currentStorage.close();
 		// Must be done last since it will result in termination of the
 		// framework active thread.
+		if (connectModules.moduleConnector instanceof FrameworkUtilHelper) {
+			FrameworkUtilHelper helper = (FrameworkUtilHelper) connectModules.moduleConnector;
+			currentExecutor.execute(() -> {
```

Comment:
Why is this done async here?

Reply:
This is so this happens after all pending task of the executor as a very last action ... because we don't wait here to the shutdown of the executor.... but maybe that is not necessary...?

Reply:
I would just do it synchronously.  If we run into an issue with that then we can understand if something else is needed.